### DISCUSSION TITLE
About Contribution to ESPResSo

### DISCUSSION THREAD

**Discussion by bishwash369** [espressomd/espresso#4641] [General]
Hello Team,
I am a Master's student and am supposed to contribute to scientific software, which is a part of my university module. I have been assigned with ESPResSo. 

I checked some of the issues on GitHub and I realized that most of them require prior knowledge of physics. I do not have experience or prior knowledge of Physics and Simulation. However, I have about 2 years of experience in coding and Software Development (using C#/ ASP.NET Core). 

Please let me know if there are any ways for me to contribute. Whether it is about changes, documentation, or adding new features.

Thanks and Regards,
Bishwash :)

**Comment by jngrad** [comment]
Hi Bishwash,

Thank you for volunteering to contribute to ESPResSo! We typically put the [CodingDay](https://github.com/espressomd/espresso/issues?q=is%3Aissue+is%3Aopen+label%3ACodingDay) label on small projects that can be completed in an afternoon. Sometimes, these tickets provide a minimal Python script that sets up the required simulation parameters to reproduce the bug or to illustrate how a missing feature is impacting the simulation, so that people can contribute even if they are not familiar with the physics behind the request.

We also organize coding days on Zoom every three months on average. During these events, people have the opportunity to get briefed on the physics by an expert, or to work in pairs by sharing their screen or using the live code feature of their code editor.

We also have a few software engineering topics, but they are long-term projects (> 3 months) that require some C++ experience.

The ESPResSo team will be back on January 9, maybe then we can think about a suitable topic for you. You could probably also meet with us directly, since you're close by. Do you have requirements in terms of a deadline or number of hours to invest?

Best regards,
JN

**Reply by bishwash369** [reply]
Hello JN,

Thank you for the reply and the briefing. I am really relieved that I can contribute without prior knowledge of physics.
I am a student at the University of Stuttgart so it would be great to meet the team in person. 

My course ends at the beginning of February, So I hope that I can start the project as soon as possible. My exams begin in mid-February, so I am hoping to finish them before then. Please let me know when we can meet and if you have any requirements for me :)

Thanks and Regards,
Bishwash

**Reply by bishwash369** [reply]
Dear JN,

@jngrad Due to time constraints I think it would be better if I go ahead with one of the CodingDay challenges. I am trying to look into the challenges but I do not fully understand the requirements. Please let me know if you and your team have any requirements in mind that I can fix. Also, it would be of great help if we could talk further about the requirements at hand.

Thanks and Regards,
Bishwash

**Comment by RudolfWeeber** [comment]
Hi Bishwash, 
unfortunately, ESPResSo is not written in the languages you listed, and most of the coding day tasks require some previous experience with the software.
After discussion with the course instructors, we'd suggest te following task, which only requires bash and language agnostic CI/CD tooling:

Based on existing work[1] which builds an Ubuntu package for ESPResSo automatically, use Github Actions[2] to automatically build an Ubuntu package repository (ppa[3]), from whcih Ubuntu users can easily install ESPResSo.
A github action for upoadign to a ppa can be found here[4].

We can set up a short webex meeting to discuss the project.

[1] https://github.com/RudolfWeeber/espresso/tree/debianization/debian and https://github.com/RudolfWeeber/espresso/tree/debianization/.github
[2] https://docs.github.com/en/actions/learn-github-actions/understanding-github-actions
[3] https://askubuntu.com/questions/71510/how-do-i-create-a-ppa
[4] https://github.com/yuezk/publish-ppa-package

Regards, Rudolf

**Reply by bishwash369** [reply]
Hi Rudolf,

Thank you for the reply and the task above. I see that I would need a launchpad account to create a PPA package which would make me the maintainer of the package. I am not sure if I am supposed to set up an account or my own or if you'd create one for me. Please let me know when you/your team are available for a Webex meeting.

Sincere Regards,
Bishwash

**Comment by RudolfWeeber** [comment]
Hi, pleasee create a Launchpad account under your own name and fork the espressomd/espresso repo. Once your work is compleed, we can switch to alaunchpad account operated by the Espresso team.

Regards, Rudolf

**Reply by bishwash369** [reply]
Dear Rudolph,

For the last two weeks, I have tried and worked on building and publishing the PPA packages via GitHub Actions. I have managed to build packages for Ubuntu versions of **Jammy, Kinetic, and Lunar**. In contrast, the builds for **Bionic and Focal** fail due to package dependencies which I have been unable to figure out. 

Please let me know if I can make any further changes.

<https://github.com/bishwash369/espresso>
<https://launchpad.net/~bishwash369/+archive/ubuntu/espressoppa/+packages>

Thanks and regards,
Bishwash
